src/ray/core_worker/store_provider/memory_store/memory_store.cc

```diff
@@ -193,22 +197,19 @@ Status CoreWorkerMemoryStore::Get(const std::vector<ObjectID> &object_ids,
           // because `object_ids` might have duplicate ids.
           ids_to_remove.insert(object_id);
         }
+        count += 1;
       } else {
         remaining_ids.insert(object_id);
       }
     }
+    RAY_CHECK(count <= num_objects);
 
     for (const auto &object_id : ids_to_remove) {
       objects_.erase(object_id);
     }
 
     // Return if all the objects are obtained.
-    if (remaining_ids.empty()) {
-      return Status::OK();
-    }
-
-    if (object_ids.size() - remaining_ids.size() >= static_cast<size_t>(num_objects)) {
-      // Already get enough objects.
+    if (remaining_ids.empty() || count >= num_objects) {
```

Comment:
```suggestion
    if (count == num_objects) {
```

from the above logic `count` cannot be larger than `num_objects`, and also it seems like it's not necessary to check if `remaining_ids.empty()`?